src/sparsezoo/__init__.py

```diff
@@ -25,3 +26,59 @@
 from .main import *
 from .models.zoo import *
 from .objects import *
+
+from .requests.base import LATEST_PACKAGE_VERSION_URL
+
+import logging
+import requests
+import threading
+import os
+
+_LOGGER = logging.getLogger(__name__)
+
+def _version_check(package_name, package_version, package_integration):
+    url = f"{LATEST_PACKAGE_VERSION_URL}?"\
+        f"packages={package_name}"\
+        f"&versions={package_version}"\
+        f"&integrations={package_integration}"
+    try:
+        response = requests.get(url)  # no token-headers required
+        response.raise_for_status()
+
+        response_json = response.json()
+        for checkedPackage in response_json["checkedPackages"]:
+            if not checkedPackage["isLatest"]:
+                _LOGGER.warning(
+                    "WARNING: "\
+                    f"You are using {checkedPackage['packageName']} "\
+                    f"version {checkedPackage['userVersion']} "\
+                    f"however version {checkedPackage['latestVersion']} "\
+                    "is available.\n"\
+                    "You should consider upgrading via executing the "\
+                    f"'pip install {checkedPackage['packageName']}'"\
+                    "command"
+                )
+                _LOGGER.warning(
+                    "WARNING: "\
+                    "To turn off version check, set an environmental variable " \
+                    "NM_VERSION_CHECK=false"\
+                )
+    except:
+        pass
```

Comment:
we should be logging this exception here as a warning rather than silently ignoring it so we can surface any issues with the API or the contained code